src/service/action-impl.js

```diff
@@ -224,12 +226,22 @@ export class ActionService {
       return;
     }
 
+    if (action.actionInfo.target === 'AMP') {
+      if (action.actionInfo.method === 'setState') {
```

Comment:
We should instead create an AMP object in `src` and register it with action service as a target. We have `ActionService.installActionHandler` for this purpose.